api/service/version_endpoint_service.go

```diff
@@ -246,6 +246,12 @@ func (k *endpointService) override(left *models.VersionEndpoint, right *models.V
 		left.EnvVars = right.EnvVars
 	}
 
+	// override secrets
+	// Configure environment variables for Pyfunc model
+	if right.Secrets != nil {
```

Comment:
Q: So if there are no secrets the new deployment will reuse the old secrets?

Reply:
It's more like if `secrets==nil` then it'll reuse old secrets. If `secrets==[]` then no secrets will be configured.